### PR TITLE
Weekly upstream merge

### PR DESCRIPTION
**Pull request by frank-emrich**
see comment below regarding a change in `compiler.rs`

### CODE REVIEW

File: crates/wasmtime/src/compiler.rs [hunk 1]
```diff
@@ -621,16 +625,21 @@ impl FunctionIndices {
...
-                            SignatureIndex::from_u32(key.index),
-                            symbol_ids_and_locs[i.unwrap_function()].1,
-                        )
+                let unique_and_sorted_sigs = translation
+                    .module
+                    .types
+                    .iter()
+                    .map(|(_, ty)| match ty {
+                        ModuleType::Function(ty) => *ty,
+                        ModuleType::Continuation(ty) => *ty,
```
Comment (frank-emrich) [inline crates/wasmtime/src/compiler.rs:634]: Not sure if this is correct.

Comment (dhil) [reply crates/wasmtime/src/compiler.rs:634]: It is correct.

Comment (dhil) [inline crates/wasmtime/src/compiler.rs:634]:
```suggestion
                        ModuleType::Function(ty) | ModuleType::Continuation(ty) => *ty,
```

**Review by dhil — APPROVED** [review]
LGTM.

**Comment by dhil** [comment]
Subsumed by #50.

**Pull request closed without merging**